From 995fd81064cfe8c150c78234bad592b1c30989f2 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Wed, 31 Jan 2018 15:58:10 +0100
Subject: [PATCH] FindQtWaylandScanner.cmake: Use qmake-query for HINT

Summary: Instead of hardcoded /usr/lib, same solution as in D9116

Test Plan: Works fine with kwayland 5.42.0.

Reviewers: #frameworks, #build_system, cgiboudeaux

Reviewed By: cgiboudeaux

Tags: #frameworks, #build_system

Differential Revision: https://phabricator.kde.org/D10201
---
 find-modules/FindQtWaylandScanner.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/find-modules/FindQtWaylandScanner.cmake b/find-modules/FindQtWaylandScanner.cmake
index 79a1b051..37ebbdba 100644
--- a/find-modules/FindQtWaylandScanner.cmake
+++ b/find-modules/FindQtWaylandScanner.cmake
@@ -86,11 +86,14 @@
 #=============================================================================
 
 include(${CMAKE_CURRENT_LIST_DIR}/ECMFindModuleHelpersStub.cmake)
+include("${ECM_MODULE_DIR}/ECMQueryQmake.cmake")
 
 ecm_find_package_version_check(QtWaylandScanner)
 
+query_qmake(qt_binaries_dir QT_INSTALL_BINS)
+
 # Find qtwaylandscanner
-find_program(QtWaylandScanner_EXECUTABLE NAMES qtwaylandscanner HINTS /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/qt5/bin/)
+find_program(QtWaylandScanner_EXECUTABLE NAMES qtwaylandscanner HINTS ${qt_binaries_dir})
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(QtWaylandScanner
